avr/cores/megacommand/CommonTools/RingBuffer.h

```diff
@@ -104,8 +109,14 @@ template <class C, int N, class T>
     overflow++;
     return false;
   }
+  if (ptr == NULL) {
   memcpy((void *)&buf[wr], (void *)c, sizeof(*c));
-  wr = RB_INC(wr);
+  }
+  else {
+  memcpy_bank1(ptr + wr, (void *)c, sizeof(*c));
```

Comment:
The behavior differs from the original.
`ptr` is `volatile uint8_t*`, but `buf` is `C[N]`.
`&buf[wr]` is `buf + sizeof(C) * wr`.